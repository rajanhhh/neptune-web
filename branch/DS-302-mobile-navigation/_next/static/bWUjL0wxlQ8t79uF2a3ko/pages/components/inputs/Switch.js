(window.webpackJsonp=window.webpackJsonp||[]).push([[68],{"5s8o":function(e,n,t){},"6gor":function(e,n,t){"use strict";n.a={DOWN:40,UP:38,LEFT:37,RIGHT:39,ENTER:13,ESCAPE:27,TAB:9,SPACE:32,BACKSPACE:8}},O94r:function(e,n,t){var c;!function(){"use strict";var t={}.hasOwnProperty;function a(){for(var e=[],n=0;n<arguments.length;n++){var c=arguments[n];if(c){var o=typeof c;if("string"===o||"number"===o)e.push(c);else if(Array.isArray(c)&&c.length){var r=a.apply(null,c);r&&e.push(r)}else if("object"===o)for(var i in c)t.call(c,i)&&c[i]&&e.push(i)}}return e.join(" ")}e.exports?(a.default=a,e.exports=a):void 0===(c=function(){return a}.apply(n,[]))||(e.exports=c)}()},cGSD:function(e,n,t){(window.__NEXT_P=window.__NEXT_P||[]).push(["/components/inputs/Switch",function(){return t("cjIA")}])},cjIA:function(e,n,t){"use strict";t.r(n),t.d(n,"meta",(function(){return v})),t.d(n,"default",(function(){return L}));var c=t("cxan"),a=t("HbGN"),o=t("ERkP"),r=t.n(o),i=t("ZVZ0"),l=t("Qi1R"),s=t("O94r"),u=t.n(s),d=(t("5s8o"),t("6gor")),p=function(){return r.a.createElement("svg",{width:"24",height:"24",viewBox:"0 0 24 24",fill:"none",xmlns:"http://www.w3.org/2000/svg"},r.a.createElement("path",{fillRule:"evenodd",clipRule:"evenodd",d:"M12 22C17.5228 22 22 17.5228 22 12C22 6.47715 17.5228 2 12 2C6.47715 2 2 6.47715 2 12C2 17.5228 6.47715 22 12 22ZM7.70712 11.2929L6.29291 12.7072L10.5 16.9143L18.2071 9.20715L16.7929 7.79293L10.5 14.0858L7.70712 11.2929Z",fill:"#fff"}))},f=function(){return r.a.createElement("svg",{width:"24",height:"24",viewBox:"0 0 24 24",fill:"none",xmlns:"http://www.w3.org/2000/svg"},r.a.createElement("path",{fillRule:"evenodd",clipRule:"evenodd",d:"M12 22C17.5228 22 22 17.5228 22 12C22 6.47715 17.5228 2 12 2C6.47715 2 2 6.47715 2 12C2 17.5228 6.47715 22 12 22ZM15.2929 16.7071L12 13.4142L8.70715 16.7071L7.29294 15.2929L10.5858 12L7.29291 8.70712L8.70712 7.29291L12 10.5858L15.2929 7.29291L16.7072 8.70712L13.4142 12L16.7071 15.2929L15.2929 16.7071Z",fill:"#fff"}))},h=function(e){var n=e.checked,t=e.onClick,c=e.className,a=e.id;return r.a.createElement("span",{className:u()("switch",{"switch--unchecked":!n,"switch--checked":n},c),onClick:t,onKeyDown:function(e){(32===e.code||e.keyCode===d.a.SPACE)&&(e.preventDefault(),t())},tabIndex:0,role:"checkbox","aria-checked":n,id:a},r.a.createElement("span",{className:"switch--thumb"},n?r.a.createElement(p,{filled:!0,size:24}):r.a.createElement(f,{filled:!0,size:24})),r.a.createElement("input",{type:"checkbox",checked:n,readOnly:!0}))};h.defaultProps={checked:!1,className:null,id:null};var w=h,m="() => {\n  const [checked, setCheck] = React.useState(false);\n\n  return <Switch checked={checked} onClick={() => setCheck(!checked)} />;\n};\n",v=(r.a.createElement,{name:"Switch"}),k={meta:v},E="wrapper";function L(e){var n=e.components,t=Object(a.a)(e,["components"]);return Object(i.a)(E,Object(c.a)({},k,t,{components:n,mdxType:"MDXLayout"}),Object(i.a)(l.b,{code:m,scope:{Switch:w},mdxType:"LiveEditorBlock"}),Object(i.a)(l.a,{componentName:"Switch",mdxType:"GeneratePropsTable"}))}L.isMDXComponent=!0}},[["cGSD",0,1,2,3,4,5]]]);