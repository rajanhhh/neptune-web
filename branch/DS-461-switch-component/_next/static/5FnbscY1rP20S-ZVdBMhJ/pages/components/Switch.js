(window.webpackJsonp=window.webpackJsonp||[]).push([[74],{"5s8o":function(e,n,t){},AJUK:function(e,n,t){(window.__NEXT_P=window.__NEXT_P||[]).push(["/components/Switch",function(){return t("zUAs")}])},O94r:function(e,n,t){var c;!function(){"use strict";var t={}.hasOwnProperty;function a(){for(var e=[],n=0;n<arguments.length;n++){var c=arguments[n];if(c){var o=typeof c;if("string"===o||"number"===o)e.push(c);else if(Array.isArray(c)&&c.length){var r=a.apply(null,c);r&&e.push(r)}else if("object"===o)for(var i in c)t.call(c,i)&&c[i]&&e.push(i)}}return e.join(" ")}e.exports?(a.default=a,e.exports=a):void 0===(c=function(){return a}.apply(n,[]))||(e.exports=c)}()},zUAs:function(e,n,t){"use strict";t.r(n),t.d(n,"meta",(function(){return w})),t.d(n,"default",(function(){return v}));var c=t("cxan"),a=t("HbGN"),o=t("ERkP"),r=t.n(o),i=t("ZVZ0"),s=t("Qi1R"),u=t("aWzz"),p=t.n(u),l=t("O94r"),h=t.n(l),d=(t("5s8o"),t("za2A")),f=function(e){var n=e.checked,t=e.onClick;return r.a.createElement("span",{className:h()("switch",{"switch--unchecked":!n,"switch--checked":n}),onClick:t,onKeyPress:t,role:"button",tabIndex:0},r.a.createElement("span",{className:"switch--thumb"},n?r.a.createElement(d.a,{filled:!0}):r.a.createElement(d.b,{filled:!0})),r.a.createElement("input",{type:"checkbox",checked:n,readOnly:!0}))};f.propTypes={checked:p.a.bool,onClick:p.a.func},f.defaultProps={onClick:function(){},checked:!1};var m=f,k="() => {\n  const [checked, setCheck] = React.useState(false);\n\n  return <Switch checked={checked} onClick={() => setCheck(!checked)} />;\n};\n",w=(r.a.createElement,{name:"Switch"}),b={meta:w},y="wrapper";function v(e){var n=e.components,t=Object(a.a)(e,["components"]);return Object(i.a)(y,Object(c.a)({},b,t,{components:n,mdxType:"MDXLayout"}),Object(i.a)(s.b,{code:k,scope:{Switch:m},mdxType:"LiveEditorBlock"}),Object(i.a)(s.a,{componentName:"Switch",mdxType:"GeneratePropsTable"}))}v.isMDXComponent=!0}},[["AJUK",0,1,2,15,3,4]]]);