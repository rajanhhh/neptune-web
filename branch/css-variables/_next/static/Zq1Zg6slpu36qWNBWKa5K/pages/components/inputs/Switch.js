(window.webpackJsonp=window.webpackJsonp||[]).push([[67],{"5s8o":function(e,n,t){},"6gor":function(e,n,t){"use strict";n.a={DOWN:40,UP:38,LEFT:37,RIGHT:39,ENTER:13,ESCAPE:27,TAB:9,SPACE:32,BACKSPACE:8}},O94r:function(e,n,t){var c;!function(){"use strict";var t={}.hasOwnProperty;function a(){for(var e=[],n=0;n<arguments.length;n++){var c=arguments[n];if(c){var o=typeof c;if("string"===o||"number"===o)e.push(c);else if(Array.isArray(c)&&c.length){var i=a.apply(null,c);i&&e.push(i)}else if("object"===o)for(var r in c)t.call(c,r)&&c[r]&&e.push(r)}}return e.join(" ")}e.exports?(a.default=a,e.exports=a):void 0===(c=function(){return a}.apply(n,[]))||(e.exports=c)}()},cGSD:function(e,n,t){(window.__NEXT_P=window.__NEXT_P||[]).push(["/components/inputs/Switch",function(){return t("cjIA")}])},cjIA:function(e,n,t){"use strict";t.r(n),t.d(n,"meta",(function(){return E})),t.d(n,"default",(function(){return b}));var c=t("cxan"),a=t("HbGN"),o=t("ERkP"),i=t.n(o),r=t("ZVZ0"),l=t("Qi1R"),s=t("aWzz"),u=t.n(s),d=t("O94r"),p=t.n(d),f=(t("5s8o"),t("6gor")),h=function(){return i.a.createElement("svg",{width:"24",height:"24",viewBox:"0 0 24 24",fill:"none",xmlns:"http://www.w3.org/2000/svg"},i.a.createElement("path",{fillRule:"evenodd",clipRule:"evenodd",d:"M12 22C17.5228 22 22 17.5228 22 12C22 6.47715 17.5228 2 12 2C6.47715 2 2 6.47715 2 12C2 17.5228 6.47715 22 12 22ZM7.70712 11.2929L6.29291 12.7072L10.5 16.9143L18.2071 9.20715L16.7929 7.79293L10.5 14.0858L7.70712 11.2929Z",fill:"#fff"}))},w=function(){return i.a.createElement("svg",{width:"24",height:"24",viewBox:"0 0 24 24",fill:"none",xmlns:"http://www.w3.org/2000/svg"},i.a.createElement("path",{fillRule:"evenodd",clipRule:"evenodd",d:"M12 22C17.5228 22 22 17.5228 22 12C22 6.47715 17.5228 2 12 2C6.47715 2 2 6.47715 2 12C2 17.5228 6.47715 22 12 22ZM15.2929 16.7071L12 13.4142L8.70715 16.7071L7.29294 15.2929L10.5858 12L7.29291 8.70712L8.70712 7.29291L12 10.5858L15.2929 7.29291L16.7072 8.70712L13.4142 12L16.7071 15.2929L15.2929 16.7071Z",fill:"#fff"}))},m=function(e){var n=e.checked,t=e.onClick,c=e.className,a=e.id;return i.a.createElement("span",{className:p()("switch",{"switch--unchecked":!n,"switch--checked":n},c),onClick:t,onKeyDown:function(e){(32===e.code||e.keyCode===f.a.SPACE)&&(e.preventDefault(),t())},tabIndex:0,role:"checkbox","aria-checked":n,id:a},i.a.createElement("span",{className:"switch--thumb"},n?i.a.createElement(h,{filled:!0,size:24}):i.a.createElement(w,{filled:!0,size:24})),i.a.createElement("input",{type:"checkbox",checked:n,readOnly:!0}))};m.propTypes={checked:u.a.bool,onClick:u.a.func.isRequired,className:u.a.string,id:u.a.string},m.defaultProps={checked:!1,className:null,id:null};var k=m,v="() => {\n  const [checked, setCheck] = React.useState(false);\n\n  return <Switch checked={checked} onClick={() => setCheck(!checked)} />;\n};\n",E=(i.a.createElement,{name:"Switch"}),C={meta:E},L="wrapper";function b(e){var n=e.components,t=Object(a.a)(e,["components"]);return Object(r.a)(L,Object(c.a)({},C,t,{components:n,mdxType:"MDXLayout"}),Object(r.a)(l.b,{code:v,scope:{Switch:k},mdxType:"LiveEditorBlock"}),Object(r.a)(l.a,{componentName:"Switch",mdxType:"GeneratePropsTable"}))}b.isMDXComponent=!0}},[["cGSD",0,1,2,3]]]);