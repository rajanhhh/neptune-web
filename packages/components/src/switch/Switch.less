@import (reference) '@transferwise/neptune-css/src/variables/neptune-tokens.less';

.switch {
  display: inline-flex;
  width: 50px;
  overflow: hidden;
  border-radius: 16px;
  padding: 2px;
  cursor: pointer;
  border: 1px solid @color-base-smoke-mid;
  &--thumb {
    display: flex;
    transition: transform cubic-bezier(0, 0.94, 0.62, 1) 350ms;
    .tw-icon {
      color: @color-base-white;
    }
  }
  &--unchecked {
    background: @color-base-smoke-mid;
    .switch--thumb {
      transform: translateX(0);
    }
  }
  &--checked {
    border: 1px solid @color-base-blue-light;
    background: @color-base-blue-light;
    .switch--thumb {
      transform: translateX(20px);
    }
  }
  input {
    opacity: 0;
    display: none;
    position: absolute;
    width: 0;
    height: 0;
    left: -100%;
  }
}
