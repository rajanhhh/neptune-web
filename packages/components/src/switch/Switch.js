import React from 'react';
import Types from 'prop-types';
import classnames from 'classnames';
import './Switch.css';
// @TODO Add this once new icons are released
// import { CheckCircle, CrossCircle } from '@transferwise/icons';
import KeyCodes from '../common/keyCodes';

// @TODO Remove this once new icons are released
const CheckCircle = () => (
  <svg width="24" height="24" viewBox="0 0 24 24" fill="none" xmlns="http://www.w3.org/2000/svg">
    <path
      fillRule="evenodd"
      clipRule="evenodd"
      d="M12 22C17.5228 22 22 17.5228 22 12C22 6.47715 17.5228 2 12 2C6.47715 2 2 6.47715 2 12C2 17.5228 6.47715 22 12 22ZM7.70712 11.2929L6.29291 12.7072L10.5 16.9143L18.2071 9.20715L16.7929 7.79293L10.5 14.0858L7.70712 11.2929Z"
      fill="#fff"
    />
  </svg>
);
const CrossCircle = () => (
  <svg width="24" height="24" viewBox="0 0 24 24" fill="none" xmlns="http://www.w3.org/2000/svg">
    <path
      fillRule="evenodd"
      clipRule="evenodd"
      d="M12 22C17.5228 22 22 17.5228 22 12C22 6.47715 17.5228 2 12 2C6.47715 2 2 6.47715 2 12C2 17.5228 6.47715 22 12 22ZM15.2929 16.7071L12 13.4142L8.70715 16.7071L7.29294 15.2929L10.5858 12L7.29291 8.70712L8.70712 7.29291L12 10.5858L15.2929 7.29291L16.7072 8.70712L13.4142 12L16.7071 15.2929L15.2929 16.7071Z"
      fill="#fff"
    />
  </svg>
);

/**
 * Switch component provides a simple switch button.
 *
 * @param {boolean} [checked=false] - boolean to set the switch status.
 * @param {function} onClick - function that manage the check logic.
 * @param {string} [id=null] - string that identifies the element.It can be use to couple the switch to its label.
 * @param {string} [className=null] - string that contains extra-classes.
 * @param {object} translations - map of keys to translated labels
 * @param {string} translations.ariaDescription - provide a description of what you are toggling
 *
 * @usage '<Switch checked={checked} onClick={() => setCheck(!checked)} id='id' className='extra-class-name' />'
 * */

const Switch = ({ checked, onClick, className, id, translations }) => {
  const handleKeyDown = (event) => {
    if (event.code === 32 || event.keyCode === KeyCodes.SPACE) {
      event.preventDefault();
      onClick();
    }
  };
  return (
    <span
      className={classnames(
        'switch',
        {
          'switch--unchecked': !checked,
          'switch--checked': checked,
        },
        className,
      )}
      onClick={onClick}
      onKeyDown={handleKeyDown}
      tabIndex={0}
      role="switch"
      aria-checked={checked}
      aria-label={translations.ariaDescription}
      id={id}
    >
      <span className="switch--thumb">
        {checked ? <CheckCircle filled size={24} /> : <CrossCircle filled size={24} />}
      </span>
      <input type="checkbox" checked={checked} readOnly />
    </span>
  );
};

Switch.propTypes = {
  checked: Types.bool,
  onClick: Types.func.isRequired,
  className: Types.string,
  id: Types.string,
  translations: Types.shape({
    ariaDescription: Types.string.isRequired,
  }).isRequired,
};
Switch.defaultProps = {
  checked: false,
  className: null,
  id: null,
};

export default Switch;
