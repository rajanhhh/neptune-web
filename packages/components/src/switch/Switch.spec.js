import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import Switch from './Switch';

import KeyCodes from '../common/keyCodes';

const props = {
  checked: true,
  onClick: jest.fn(),
  id: 'id',
  className: 'a-class-name',
  translations: { ariaDescription: 'a label' },
};

describe('Switch', () => {
  it('renders component when checked', () => {
    const { asFragment } = render(
      <Switch
        checked={props.checked}
        onClick={props.onClick}
        className={props.className}
        id={props.id}
        translations={props.translations}
      />,
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it('renders component when unchecked', () => {
    const { asFragment } = render(
      <Switch
        checked={false}
        onClick={props.onClick}
        className={props.className}
        id={props.id}
        translations={props.translations}
      />,
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it('calls onClick when user press space key', () => {
    const { getAllByRole } = render(
      <Switch
        checked={props.checked}
        onClick={props.onClick}
        className={props.className}
        id={props.id}
        translations={props.translations}
      />,
    );

    const input = getAllByRole('checkbox')[0];
    fireEvent.keyDown(input, { key: '33', keyCode: KeyCodes.ENTER });
    expect(props.onClick).not.toHaveBeenCalled();
    fireEvent.keyDown(input, { key: '32', keyCode: KeyCodes.SPACE });
    expect(props.onClick).toHaveBeenCalled();
  });
});
