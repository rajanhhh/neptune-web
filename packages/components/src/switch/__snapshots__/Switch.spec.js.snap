// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Switch renders component when checked 1`] = `
<DocumentFragment>
  <span
    aria-checked="true"
    aria-label="a label"
    class="switch switch--checked a-class-name"
    id="id"
    role="switch"
    tabindex="0"
  >
    <span
      class="switch--thumb"
    >
      <svg
        fill="none"
        height="24"
        viewBox="0 0 24 24"
        width="24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          clip-rule="evenodd"
          d="M12 22C17.5228 22 22 17.5228 22 12C22 6.47715 17.5228 2 12 2C6.47715 2 2 6.47715 2 12C2 17.5228 6.47715 22 12 22ZM7.70712 11.2929L6.29291 12.7072L10.5 16.9143L18.2071 9.20715L16.7929 7.79293L10.5 14.0858L7.70712 11.2929Z"
          fill="#fff"
          fill-rule="evenodd"
        />
      </svg>
    </span>
    <input
      checked=""
      readonly=""
      type="checkbox"
    />
  </span>
</DocumentFragment>
`;

exports[`Switch renders component when unchecked 1`] = `
<DocumentFragment>
  <span
    aria-checked="false"
    aria-label="a label"
    class="switch switch--unchecked a-class-name"
    id="id"
    role="switch"
    tabindex="0"
  >
    <span
      class="switch--thumb"
    >
      <svg
        fill="none"
        height="24"
        viewBox="0 0 24 24"
        width="24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          clip-rule="evenodd"
          d="M12 22C17.5228 22 22 17.5228 22 12C22 6.47715 17.5228 2 12 2C6.47715 2 2 6.47715 2 12C2 17.5228 6.47715 22 12 22ZM15.2929 16.7071L12 13.4142L8.70715 16.7071L7.29294 15.2929L10.5858 12L7.29291 8.70712L8.70712 7.29291L12 10.5858L15.2929 7.29291L16.7072 8.70712L13.4142 12L16.7071 15.2929L15.2929 16.7071Z"
          fill="#fff"
          fill-rule="evenodd"
        />
      </svg>
    </span>
    <input
      readonly=""
      type="checkbox"
    />
  </span>
</DocumentFragment>
`;
